src/app/icd/BUILD.gn

```diff
@@ -16,11 +16,20 @@ import("//build_overrides/chip.gni")
 import("icd.gni")
 
 # ICD Server sources and configurations
+config("notifier-config") {
+  defines =
+      [ "ICD_MAX_NOTIFICATION_SUBSCRIBERS=${icd_max_notification_subscribers}" ]
+}
 
 source_set("observer-srcs") {
   sources = [ "ICDStateObserver.h" ]
 }
 
+source_set("notifier-srcs") {
```

Comment:
nit: I believe the `-srcs` suffix here should not be needed - generally these targets types are transparent (static_libs or source_sets) and we do not use this pattern anywhere else.